### DISCUSSION TITLE
Google Summer of Code 2023 CNCF Project Proposals

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#777] [Announcements]
Hello!

CNCF has been participating in the Google Summer of Code during the [last few years](https://www.cncf.io/blog/2021/11/01/16-cncf-interns-graduated-from-google-summer-of-code-gsoc-2021/), and now we are ready to start the 2023 program cycle.
 
All CNCF projects are eligible to apply to participate in the Google Summer of Code. Please start working on the project ideas, and submit them via pull request: https://github.com/cncf/mentoring/blob/main/summerofcode/2023.md#project-ideas. 

The deadline to submit project ideas is March 19, 2023 (though we'd be happy for project ideas to come in sooner than that if possible).

[Google Summer of Code 2023 Timeline](https://developers.google.com/open-source/gsoc/timeline)

Thanks!

**Comment by aliok** [comment]
Thanks for organizing GSoC@CNCF @nate-double-u , looking forward to it! I've even sent a PR for Knative ideas already!

However, I have something to discuss.

> The deadline to submit project ideas is February 7, 2023.

Project ideas can be added until the very last minute before the contributor applications begin.

We had the same discussion last year: https://github.com/cncf/mentoring/pull/560#issuecomment-1062240768 
I know it was a long time ago :)

However, I must say that, there must be some ideas in the ideas page so that Google can see that we can provide ideas and mentorship to potential GSoC contributors.

**Reply by nate-double-u** [reply]
Thanks, @aliok, for the project ideas PR and for the reminder of our conversation last year.

You may be right about the project idea deadline; I'll double-check.

Regardless, it's important to get some project ideas together as soon as possible because the list of projects is included in the org application.

**Reply by nate-double-u** [reply]
I think you're right. Cross checking (again, I'm sad i got confused by this two years running) it looks like the actual cutoff date would be [March 19, 2023](https://developers.google.com/open-source/gsoc/timeline#february_22_-_march_19).

**Reply by aliok** [reply]
Thanks for checking it @nate-double-u . I get confused about so many GSoC things myself, while running it for JBoss community, so I know 😃

**Reply by 9avneet** [reply]
@nate-double-u is  there any other projects that are included in gsoc 23 or in 2023 cncf are continuing with four project only.can you clarify a little bit?? please.

**Reply by nate-double-u** [reply]
Hi @9avneet. I hope there will be, but the deadline for proposals from CNCF projects is March 19, 2023, so there is a lot of time yet for more proposals to come in.

Take a look at the [Google Summer of Code 2023 Timeline](https://developers.google.com/open-source/gsoc/timeline) to see which phase we are in.

**Reply by 9avneet** [reply]
@nate-double-u okay I got it.

**Comment by aamirpatel** [comment]
ok, Thanks @nate-double-u .

**Comment by DavidPHirsch** [comment]
Hi @nate-double-u ,
Thank you for organizing this!
we would be interested in participating, with Keptn and OpenFeature. 
Just have 2 questions:
1) Is there then some selection of the proposals? 
2) Is the deadline still the 19th of March?

**Reply by aliok** [reply]
@DavidPHirsch 

Thanks for your interest in GSoC at CNCF!

> Is there then some selection of the proposals?

Yes. Google will give organizations some limited number of slots. These numbers are currently unknown (in fact, there's no guarantee that Google will accept any organization, not limited to CNCF).

To give you more information: Later in the process, potential mentees will submit proposals to GSoC system. Organizations (CNCF in this case) will have to make some selections after mentee project proposals are submitted. This selection will be based on various things like, contributor<>community interaction, contributor's project proposal quality, proposal impact, etc.

> Is the deadline still the 19th of March?

Yes it is. However, it is probably a little too late for any contributor to see your project idea and interact with you and your community before submitting a proposal. Recommended latest date for adding ideas is end of February to give potential mentees some time to explore your project idea and the community.

**Comment by beeme1mr** [comment]
Hi @aliok, we had someone reach out via email to let us know they were interested in our project proposal. Since the proposal hasn't been officially accepted as a GSoC project, how would you recommend we proceed? FYI @DavidPHirsch.

**Reply by aliok** [reply]
hi @beeme1mr

I would guide them to start working towards the proposed project already. Advice them to start exploring the project, try tutorials, try to fix some `good-first-issue`s, etc. In the end, you and CNCF org admins will want to accept a contributor to GSoC program only if a candidate showed good community interaction and also showed that they are capable of doing the necessary tasks (of course acceptance criteria is not limited to these 2). It is very good that they reached out early. They would have more time to demonstrate these.
You can directly share the message above :)

BTW, even if you really liked the candidate and you want to work with them, you cannot tell them that you want to accept their application when they apply. It is not only up to the mentor anyway. CNCF GSoC admins and Google also has a say, based on the slots CNCF gets.
Google wants to keep the acceptance information private to the community (CNCF in this case). On [May 4](https://developers.google.com/open-source/gsoc/timeline#may_4_-_1800_utc), Google will announce the accepted contributors.

cc @nate-double-u if you want to add something or want to correct something

**Comment by amittri1025** [comment]
Hello @aliok , I am linux enthusiast want to be part of GSOC 2023, things get a little confusing here, idk what project proposal are, ig we send it in end, right ? first we contribute ?
So, I should contribute in this repository before March 19

**Reply by aliok** [reply]
@amittri1025 We encourage folks to start contributing early to achieve these:

* Having a better understanding of the project
* Showing the mentor and the community that you are capable of finishing the project
* Finding out yourself if you will actually like working on the project

Proposals will come later. Google will announce today at 18:00 UTC if CNCF is an accepted organization or not. If CNCF gets accepted, then candidate mentees will start submitting proposals. You might want to check out GSoC website for more information about all of these, including the timeline and also about what a proposal should look like.

**Comment by nate-double-u** [comment]
We have been accepted as an org for GSoC 2023! Thanks for all the proposals everyone!

For more information on what happens next, please see discussion https://github.com/cncf/mentoring/discussions/848

**Reply by bwplotka** [reply]
Can we add more proposals as mentioned in this discussion (deadline March 19th?)

**Reply by bwplotka** [reply]
Also asked https://github.com/cncf/mentoring/discussions/848#discussioncomment-5194494 for clarity as "adding proposal" might mean two things (:

**Reply by nate-double-u** [reply]
Yup, the deadline is still March 19th -- we suggest adding proposals sooner than that though.

See @aliok's post: https://github.com/cncf/mentoring/discussions/777#discussioncomment-4958312

**Comment by Pratham-19** [comment]


**Reply by aliok** [reply]
@Pratham-19 

I would recommend these:
- Check out the upstream issues listed in the idea list
- Go and check out the README.md files for the projects
- See the contribution guidelines for them.
- Find out how you can reach out to the communities of them
- Read more about the projects
- Try tutorials of the projects

In summary, I would recommend doing some research and engaging with the community and the mentors.
